vignettes/occds.Rmd

```diff
@@ -240,25 +236,25 @@ ex_single <- derive_vars_dtm(
   new_vars_prefix = "EXST",
   flag_imputation = "none"
 )
-adae <- adae %>%
-  derive_var_last_dose_date(
-    ex_single,
-    filter_ex = (EXDOSE > 0 | (EXDOSE == 0 & grepl("PLACEBO", EXTRT))) &
-      !is.na(EXSTDTM),
-    dose_date = EXSTDTM,
-    analysis_date = ASTDT,
-    single_dose_condition = (EXSTDTC == EXENDTC),
-    new_var = LDOSEDTM,
-    output_datetime = TRUE
-  )
+
+# adae <- derive_vars_joined(
+#     adae,
+#     ex_single,
+#     by_vars = exprs(STUDYID, USUBJID),
+#     new_vars = exprs(LDOSEDT = EXSTDTM),
+#     join_vars = exprs(EXSTDTM),
+#     filter_add = (EXDOSE > 0 | (EXDOSE == 0 & grepl("PLACEBO", EXTRT))) & !is.na(EXSTDTM),
+#     filter_join = EXSTDTM <= ASTDTM,
+#     mode = "last"
+# )
```

Comment:
@bundfussr having some trouble reworking this example off the cuff, not sure if you have a quick draft solution, and it poses a good point for the deprecation of these functions, do the old parameters always map perfectly to the new `derive_vars_joined()` and do we still plug in `derive_vars_joined()` inside the function or just leave a warning

Reply:
The `derive_vars_joined()` call looks good to me except that the new variable should be named `LDOSEDT` instead of `LDOSEDTM`. Did I miss something?

However, I think we should implement the allow expression issue (https://github.com/pharmaverse/admiral/issues/1727)  before deprecating the `derive_var_last_*()` functions. For deprecating `derive_var_last_dose_grp()` we need https://github.com/pharmaverse/admiral/issues/1727. And in this example the pre-processing step could be replaced by an expression in the `derive_vars_joined()` call (`new_vars = exprs(LDOSEDTM = convert_dtc_to_dtm(EXSTDTC))`).

Reply:
I see. Adding `order = exprs(EXSTDTM)` should fix it. Sorry missed that.